addons/binding/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/ws/http/IhcHttpsClient.java

```diff
@@ -0,0 +1,141 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ihc.internal.ws.http;
+
+import java.io.IOException;
+import java.net.SocketTimeoutException;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.NoHttpResponseException;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.util.EntityUtils;
+import org.openhab.binding.ihc.internal.ws.exeptions.IhcExecption;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Simple HTTP Client for IHC / ELKO LS Controller connection purposes.
+ *
+ * @author Pauli Anttila - Initial contribution
+ */
+public abstract class IhcHttpsClient {
+
+    private final Logger logger = LoggerFactory.getLogger(IhcHttpsClient.class);
+
+    private IhcConnectionPool ihcConnectionPool;
+    private HttpClient client;
+    private HttpPost postReq;
+    private AtomicInteger counter = new AtomicInteger();
+
+    public IhcHttpsClient(IhcConnectionPool ihcConnectionPool) {
+        this.ihcConnectionPool = ihcConnectionPool;
+    }
+
+    /**
+     * Init HTTP connection.
+     *
+     * @param url Url to connect.
+     */
+    private void initConnection(String url) throws IhcExecption {
+        if (client == null) {
+            client = ihcConnectionPool.getHttpClient();
+        }
+        postReq = new HttpPost(url);
+    }
+
+    /**
+     * Send HTTP request and wait response from the server.
+     *
+     */
+    public String sendQuery(String url, Map<String, String> requestProperties, String query, int timeout)
+            throws IhcExecption {
+        initConnection(url);
+        setRequestProperty(requestProperties);
+        try {
+            return sendQ(query, timeout);
+        } catch (NoHttpResponseException e) {
+            try {
+                logger.debug("No response received, resend query");
+                return sendQ(query, timeout);
+            } catch (IOException ee) {
+                throw new IhcExecption(ee);
+            }
+        } catch (SocketTimeoutException e) {
+            try {
+                logger.debug("Timeout received, resend query");
+                return sendQ(query, timeout);
+            } catch (IOException ee) {
+                throw new IhcExecption(ee);
+            }
+        } catch (IOException e) {
+            throw new IhcExecption(e);
+        }
+    }
+
+    private String sendQ(String query, int timeout)
+            throws ClientProtocolException, IOException, NoHttpResponseException {
+        postReq.setEntity(new StringEntity(query, "UTF-8"));
```

Comment:
I would prefer to use `StandardCharsets.UTF_8.name()` [see](https://docs.oracle.com/javase/7/docs/api/java/nio/charset/StandardCharsets.html#UTF_8) 

And you can import the `UTF_8` to make it a little more compact.

Reply:
Done